### PR TITLE
Pack NuGet.Server from .nuspec instead of .csproj to control content subdirectory (net46)

### PR DESCRIPTION
**Pull request by joelverhagen**
.nupkg directory structure change:
```
content\Default.aspx                -->   content\net46\Default.aspx
content\favicon.ico                 -->   content\net46\favicon.ico
content\Packages                    -->   content\net46\Packages
content\Web.config                  -->   content\net46\Web.config
content\web.config.install.xdt      -->   content\net46\web.config.install.xdt
content\DataServices\Packages.svc   -->   content\net46\DataServices\Packages.svc
content\DataServices\Routes.cs.pp   -->   content\net46\DataServices\Routes.cs.pp
content\Packages\Readme.txt         -->   content\net46\Packages\Readme.txt
lib\net46\NuGet.Server.dll          -->   lib\net46\NuGet.Server.dll
```

Needed to pack a .nuspec for NuGet.Server instead of a .csproj so we can set the content files to go to `content/net46`.

Fixes https://github.com/NuGet/NuGetGallery/issues/3257

/cc @emgarten @xavierdecoster @maartenba

### CODE REVIEW

**Comment by joelverhagen** [comment]
Note that this is merging to `master` instead of `dev`, so we can release this without worrying about all of the WCF to WebAPI migrations issues.

**Comment by joelverhagen** [comment]
After this is merged, I plan on releasing NuGet.Server 2.11.3.

File: src/CommonAssemblyInfo.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 [assembly: ComVisible(false)]
 
-[assembly: AssemblyVersion("2.11.2.0")]
-[assembly: AssemblyFileVersion("2.11.2.0")]
+[assembly: AssemblyVersion("2.11.3.0")]
+[assembly: AssemblyFileVersion("2.11.3.0")]
```
Comment (emgarten) [inline src/CommonAssemblyInfo.cs]: Should nuget.server just move to 2.14.0 to match nuget.core?  These packages use to be released together. Either way is fine, it just seems like we could simplify things here.

File: src/NuGet.Server/NuGet.Server.nuspec [hunk 2]
```diff
@@ -1,23 +1,46 @@
...
+      <frameworkAssembly assemblyName="System.ServiceModel.Activation" targetFramework="net46" />
+      <frameworkAssembly assemblyName="System.Data.Services" targetFramework="net46" />
     </frameworkAssemblies>
     <dependencies>
-      <dependency id="NuGet.Core" version="[2.11.1,)" />
-      <dependency id="Microsoft.Web.Xdt" version="[2.1.1,)" />
+      <group targetFramework="net46">
+        <dependency id="NuGet.Core" version="[2.11.1,)" />
+        <dependency id="Microsoft.Web.Xdt" version="[2.1.1,)" />
+        <dependency id="Newtonsoft.Json" version="8.0.3" />
```
Comment (emgarten) [inline src/NuGet.Server/NuGet.Server.nuspec]: Any reason not to go to 9.0.1?

File: src/NuGet.Server/NuGet.Server.nuspec [hunk 3]
```diff
@@ -1,23 +1,46 @@
...
+    <file src="DataServices\Packages.svc" target="content\net46\DataServices" />
+    <file src="DataServices\Routes.cs.pp" target="content\net46\DataServices" />
+    <file src="Packages\Readme.txt" target="content\net46\Packages" />
+    <file src="Default.aspx" target="content\net46" />
+    <file src="favicon.ico" target="content\net46" />
+    <file src="web.config.install.xdt" target="content\net46" />
+    <file src="Web.config" target="content\net46" />
+    
+    <!-- src -->
+    <file src="**\*.cs" target="src" exclude="**\obj\**\*.cs" />
```
Comment (emgarten) [inline src/NuGet.Server/NuGet.Server.nuspec]: why source? shouldn't this only be in the symbol package? or is nuget.server special?

File: src/NuGet.Server/NuGet.Server.nuspec [hunk 4]
```diff
@@ -1,23 +1,46 @@
...
+        <dependency id="Newtonsoft.Json" version="8.0.3" />
+        <dependency id="RouteMagic" version="1.3" />
+        <dependency id="WebActivatorEx" version="2.1.0" />
+      </group>
     </dependencies>
   </metadata>
+  <files>
+    <!-- lib\net46 -->
+    <file src="bin\NuGet.Server.dll" target="lib\net46" />
+    <file src="bin\NuGet.Server.pdb" target="lib\net46" />
```
Comment (emgarten) [inline src/NuGet.Server/NuGet.Server.nuspec]: this needs the pdb?

File: src/CommonAssemblyInfo.cs (hunk 1 above)
Comment (joelverhagen) [inline src/CommonAssemblyInfo.cs]: Done.

File: src/NuGet.Server/NuGet.Server.nuspec (hunk 2 above)
Comment (joelverhagen) [inline src/NuGet.Server/NuGet.Server.nuspec]: Done.

**Comment by maartenba** [comment]
LGTM!

**Pull request merged**